libafl_targets/src/sancov_8bit.rs

```diff
@@ -19,3 +38,270 @@ pub extern "C" fn __sanitizer_cov_8bit_counters_init(start: *mut u8, stop: *mut
         ));
     }
 }
+
+#[must_use]
+/// Create a new [`CountersMultiMapObserver`] of the [`COUNTERS_MAP`].
+///
+/// This is a special [`MultiMapObserver`] for the [`COUNTERS_MAP`] and may be used when
+/// 8-bit counters are used for `SanitizerCoverage`. You can utilize this observer in a
+/// [`HitcountsIterableMapObserver`] like so:
+///
+/// ```rust,ignore
+/// use libafl::{
+///     observers::HitcountsIterableMapObserver,
+///     feedbacks::MaxMapFeedback,
+/// };
+/// use libafl_targets::sancov_8bit::counters_maps_observer;
+///
+/// let counters_maps_observer = unsafe { counters_maps_observer("counters-maps") };
+/// let counters_maps_hitcounts_observer = HitcountsIterableMapObserver::new(counters_maps_observer);
+/// let counters_maps_feedback = MaxMapFeedback::new(&counters_maps_hitcounts_observer);
+/// ```
+///
+/// # Safety
+///
+/// This function instantiates an observer of a `static mut` map whose contents are mutated by
+/// `SanitizerCoverage` instrumentation. This is unsafe, and data in the map may be mutated from
+/// under us at any time. It should never be assumed constant.
+pub unsafe fn counters_maps_observer(name: &'static str) -> CountersMultiMapObserver<false> {
+    CountersMultiMapObserver::new(name)
+}
+
+/// The [`CountersMultiMapObserver`] observes all the counters that may be set by
+/// `SanitizerCoverage` in [`COUNTERS_MAPS`]
+#[derive(Serialize, Deserialize, Debug)]
+#[allow(clippy::unsafe_derive_deserialize)]
+pub struct CountersMultiMapObserver<const DIFFERENTIAL: bool> {
```

Comment:
The observers should live in the main lib, not in libafl_targets (I think)

Reply:
Ah I see, I thought the dependency went the other way.
But in this case, maybe we should hide additional observers behind a `observers` flag? Dragging in `intervaltree` and `ahash` for this seems excessive for most usecases, what do you think?

Reply:
Yeah `libafl_targets` is already full of feature flags, no biggie to add another :)

I pushed that change up adding an `observers` flag and moved the observer into there.

Checking over the source, I notice that `cmplog.rs` has `CmpLogObserver` and `coverage.rs` has `DifferentialAFLMapSwapObserver` (although it is already behind the `"pointer_maps"` flag). Probably not be worth gating the `CmpLogObserver` since it doesn't add any dependencies, but it might be worth gating all three the same way to trim down static library size and make things consistent.

One more note is that `DifferentialAFLMapSwapObserver` uses `serde`, but it's the only thing other than `CountersMapsObserver` that does. We should probably make serde an optional dependency as well (we could gate it behind `"observers"` if we add it as a feature gate for all the observers in the crate), because it has the `"alloc"` feature enabled which makes it not possible to use `libafl_targets` in `no_std` currently.

Reply:
Sounds good. FWIW using alloc in no_std is totally possible. You just need to define an allocator.

Reply:
Cool, today I learned!